chatkit-core/src/main/kotlin/com/pusher/chatkit/state/AuxiliaryState.kt

```diff
@@ -0,0 +1,18 @@
+package com.pusher.chatkit.state
+
+import com.pusher.chatkit.rooms.state.JoinedRoomsState
+
+internal data class AuxiliaryState(
+    val version: Int,
+    val joinedRoomsReceived: ReducerLastChange<JoinedRoomsState>
+)
+
+internal data class ReducerLastChange<T>(val version: Int, val data: T)
+
+internal object AuxiliaryStateUtil {
```

Comment:
no need for the object, in Kotlin `fun`s can be top-level

Reply:
👍 have made this not an internal object.